include/mbedtls/debug.h

```diff
@@ -108,16 +108,16 @@
  *
  * This module provides debugging functions.
  */
-#if (defined(__MINGW32__) && __USE_MINGW_ANSI_STDIO == 0) || (defined(_MSC_VER) && _MSC_VER < 1800)
+#if defined(__MINGW32__) || (defined(_MSC_VER) && _MSC_VER < 1900)
```

Comment:
Does modern mingw still need the nonstandard definitions?

Note that in 3.6, we already require Visual Studio 2017 or above. So in 3.6 and development, the only reason to still support VS 2013 is our own CI scripts, which test with VS 2013 because that's what we want on 2.28. But 2.28 is reaching EOL so that soon won't be a concern anymore.

Reply:
>Does modern mingw still need the nonstandard definitions?

That's what it looks like, at least for the combination of Mingw-w64 8.3.0 and vsnprintf. We have the same check for MinGW in [platform.h](https://github.com/Mbed-TLS/TF-PSA-Crypto/blob/7d941e84a5b5c77f642186075ef45b3cc3214d57/drivers/builtin/include/mbedtls/platform.h#L52-L55)